src/main/java/ui/components/pickers/LabelPickerState.java

```diff
@@ -53,37 +53,27 @@ private final void update(String userInput) {
     }
 
     /**
-     * Updates current state if there is at least one matching label based on the 
-     * given keyword
-     * @param keyword
-     */
-    private final void updateIfMatchesLabel(String keyword) {
-        if (TurboLabel.hasMatchedLabel(allLabels, keyword)) {
-            updateAssignedLabels(TurboLabel.getFirstMatchingTurboLabel(allLabels, keyword));
-        }
-    }
-
-    /**
      * Updates assignedLabels based on properties of a label 
      * @param label
      */
-    public final void updateAssignedLabels(TurboLabel label) {
-        String labelName = label.getFullName();
+    public final void updateAssignedLabels(Optional<TurboLabel> label) {
+        if (!label.isPresent()) return;
```

Comment:
Hmm, these two lines are a bit dodgy. Maybe this method shouldn't be called instead of passing a null value and having it be a no-op?